controller/state/statedb.go

```diff
@@ -632,27 +647,25 @@ func (s *stateDB) countTasks(ctx context.Context) (int, error) {
 
 // GetHead gets the latest record update from the controller.
 func (s *stateDB) GetHead(ctx context.Context) (tasks.RecordUpdate, error) {
-	tx, err := s.db().Begin()
-	if err != nil {
-		return nil, err
-	}
-
-	var c string
-	if err := tx.QueryRowContext(ctx, queryHeadSQL, LATEST_UPDATE, "").Scan(&c); err != nil {
-		return nil, err
-	}
-	cidLink, err := cid.Decode(c)
-	if err != nil {
-		return nil, err
-	}
-
-	loader := txContextLoader(ctx, tx)
 	na := tasks.Type.RecordUpdate.NewBuilder()
-	if err = (cidlink.Link{Cid: cidLink}).Load(ctx, ipld.LinkContext{}, na, loader); err != nil {
-		return nil, err
-	}
+	err := s.transact(ctx, func(tx *sql.Tx) error {
+		var c string
+		err := tx.QueryRowContext(ctx, queryHeadSQL, LATEST_UPDATE, "").Scan(&c)
+		if err != nil {
+			return err
+		}
+		cidLink, err := cid.Decode(c)
+		if err != nil {
+			return err
+		}
 
-	if err := tx.Commit(); err != nil {
+		loader := txContextLoader(ctx, tx)
+		if err = (cidlink.Link{Cid: cidLink}).Load(ctx, ipld.LinkContext{}, na, loader); err != nil {
+			return err
+		}
```

Comment:
this can lead to a sub-txn, and should be done outside of this txn, i think?

Reply:
I don't think so no? cause the txContextLoader takes the passed in transaction.